src/mscorlib/src/System/IO/MemoryStream.cs

```diff
@@ -402,31 +402,42 @@ public override Task<int> ReadAsync(Byte[] buffer, int offset, int count, Cancel
             return _buffer[_position++];
         }
 
+        public override void CopyTo(Stream destination, int bufferSize)
+        {
+            // Since we did not originally override this method, validate the arguments
+            // the same way Stream does for back-compat.
+            ValidateCopyToArguments(destination, bufferSize);
 
-        public override Task CopyToAsync(Stream destination, Int32 bufferSize, CancellationToken cancellationToken) {
-
-            // This implementation offers beter performance compared to the base class version.
-
-            // The parameter checks must be in sync with the base version:
-            if (destination == null)
-                throw new ArgumentNullException("destination");
+            // If we have been inherited into a subclass, the following implementation could be incorrect
+            // since it does not call through to Read() or Write() which a subclass might have overriden.  
```

Comment:
Nits:
- the base CopyTo doesn't call Write on this stream, so it's not really relevant here
- overriden => overridden